### ISSUE TITLE
How to update the deterministic integration tests for the batched drivers

### ISSUE THREAD

**Issue by ye-luo** [QMCPACK/qmcpack#2046]
This is what I found in #2043
I tried switching to the batched drivers by default to see what happens with the current deterministic integration tests. LiH dimer passes out-of-the box. However
In calculations with PBC, the non-batched driver use makeMoveAndCheck which reject moves with displacement larger than half of the box while the batched driver doesn't use check since it complicates the driver a lot and doesn't seem helpful overall. This check doesn't impact large system with big boxes but affect small system. Our existing 1 walker deterministic VMC tests have small cell and large time step and thus get affected. A safe route to me is updating current tests reference with smaller time step still based on non-batched drivers and making sure trajectors are not affected by the check. Then we switch to batched drivers without updating the reference.
Any suggestions or concerns?

**Comment by prckent** [comment]
We should first discuss why we added makeMoveAndCheck to the legacy driver... good algorithm comes ahead of testing. For example, it seems like a nice check on crazy moves if the drift vector is misbehaving. We could make totally new deterministic tests for the batched driver &/or remove makeMoveAndCheck with PBC in the legacy driver...

It is very good news that LiH dimer passes.

**Comment by ye-luo** [comment]
I believe it was introduced for stopping crazy moves but I believe it can also burn users.

1. I noticed that with a particular seed, the LiH solid short test could fail due to too restrictive checks and proposed steps keep getting rejected and thus wrong results.
2. If people do want large moves, we can get spurious and biased results by the checks.
3. we do have drift modifiers and crazy moves will anyway be limited.
4. This check was also used to force electron inside box when forcing spline orbitals with open BC. Indeed, this is not necessary if we define orbitals in the whole space by putting zero value if going out of the box, the WF ratio is zero and the usually accept/rejection based on the norm of ratio anyway rejects the move.

**Comment by prckent** [comment]
Agreed. (If someone finds problems with too large moves in future we should do the simple thing and introduce an option to limit moves to some input size and not have a hidden restriction.)
